ph5/clients/ph5availability.py

```diff
@@ -473,22 +521,70 @@ def get_availability(self, station='*', location='*',
                         ph5_seed_station, ph5_loc, ph5_channel = ret
                         ph5_das = st['das/serial_number_s']
                         ph5_channum = st['channel_number_i']
-                        ph5_sample_rate = self.get_sample_rate(st)
-
-                        times = self.ph5.get_availability(
-                            ph5_das, ph5_sample_rate, ph5_channum,
-                            starttime, endtime)
+                        ph5_start_epoch = st['deploy_time/epoch_l']
+                        ph5_stop_epoch = st['pickup_time/epoch_l']
+                        Das_t = self.ph5.query_das_t(
+                            ph5_das,
+                            chan=ph5_channum,
+                            start_epoch=ph5_start_epoch,
+                            stop_epoch=ph5_stop_epoch)
+                        for das in Das_t:
+                            # Does Array.sr == DAS.sr? If so use sr
+                            if das['sample_rate_i'] == st['sample_rate_i']:
+                                samplerate_return = das['sample_rate_i']
+                                ph5_sr = das['sample_rate_i']
+                                times = self.ph5.get_availability(ph5_das,
+                                                                  ph5_sr,
+                                                                  ph5_channum,
+                                                                  starttime,
+                                                                  endtime)
+                                empty_times = False
+                            else:
+                                continue
+                        if empty_times is True:
+                            for i, das in enumerate(Das_t):
+                                # IF DAS.SR != Array.SR, USe DAS.SR if match
+                                # Checks to see if all DAS tables have same SR
+                                sr_prev = Das_t[i-1]['sample_rate_i']
+                                if das['sample_rate_i'] != sr_prev:
+                                    sr_mismatch = True
+                            if sr_mismatch is True:
+                                # Else throw warning and fail
+                                LOGGER.error('DAS and Array Table sample' +
+                                             ' rates do not match, DAS table' +
+                                             ' sample rates do not match.' +
+                                             ' Data must be updated.')
+                            else:
+                                # Uses SR if consistent
+                                samplerate_return = das['sample_rate_i']
```

Comment:
I tried to run the command ph5availability, it created error which is at this line when I removed try ... except: `UnboundLocalError: local variable 'das' referenced before assignment`.

Please consider test for main for this command and ph5toms as well.

Reply:
I fixed this error by adding try except blocks as suggested. 

```
                            try:
                                if sr_mismatch is True:
                                    LOGGER.error('DAS and Array Table sample' +
                                                 ' rates do not match, DAS' +
                                                 ' table sample rates do not' +
                                                 ' match. Data must be'
                                                 ' updated.')
                                else:
                                    # Uses SR if consistent
                                    samplerate_return = das['sample_rate_i']
                                    psr = das['sample_rate_i']
                                    early, end = self.ph5.get_extent(ph5das,
                                                                     chanum,
                                                                     psr,
                                                                     starttime,
                                                                     endtime
                                                                     )
                            except(UnboundLocalError):
                                continue
```

This, or similar, was included in PH5/ph5/clients/ph5availability/get_availability_extent() and PH5/ph5/clients/ph5availability/get_availability(), PH5/ph5/clients/ph5toms/create_trace()